cpp/src/lance/encodings/plain.cc

```diff
@@ -37,14 +37,37 @@ ::arrow::Result<int64_t> PlainEncoder::Write(std::shared_ptr<::arrow::Array> arr
   ARROW_ASSIGN_OR_RAISE(auto value_offset, out_->Tell());
   // TODO: support more types.
   switch (data_type->id()) {
+    case ::arrow::Type::INT8:
```

Comment:
Is there much value to creating a macro to generate each case? Like WRITE_ARRAY(ArrowType, ArrowArrayType) ?